scripts/ci/update_lf_base_dependency.py

```diff
@@ -0,0 +1,61 @@
+import os
+import sys
+import re
+
+import packaging.version
+
+BASE_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), "../.."))
+
+
+def update_base_dep(pyproject_path: str, new_version: str) -> None:
+    """Update the langflow-base dependency in pyproject.toml."""
+    filepath = os.path.join(BASE_DIR, pyproject_path)
+    with open(filepath, "r") as file:
+        content = file.read()
+
+    # Regex to match the langflow-base dep line under [tool.poetry]
+    # NOTE: this functions similarly to `update_dependencies.py`.
+    # The order of operations for these scripts is more complex than it needs to be;
+    # this is a reminder to revisit this process.
+    # Currently, the process is as follows:
+    # 1. nightly-build workflow updates all names, version, and the langflow-base
+    #    dependency in pyproject.toml, and commits the changes and creates a tag.
+    # 2. release-nightly workflow runs, which calls update_dependencies.py to update
+    #   the langflow-base dependency in pyproject.toml _again_. This is redundant, but
+    #   necessary because the release workflow relies on that script to update the base
+    #   dependency.
+    pattern = re.compile(r'langflow-base = \{ path = "\./src/backend/base", develop = true \}')
```

Comment:
This is worth noting. Didn't want to make changes to the release workflow in this PR since it's already a mess, but we can simplify some of this.

Reply:
By using uv instead of poetry? hahahah 

Just kidding.